### DISCUSSION TITLE
Enum controls/objects properties?

### DISCUSSION THREAD

**Discussion by EduardoVB** [twinbasic/twinbasic#305] [Ideas]
I was thinking that being able to walk the properties and get/assign their values by their property name could be useful.

Something like 

`SomeControl.Properties("Caption") = NewCaption`

and

```
Dim p As Property

For Each p in SomeControl.Properties
    Debug.Print p.Name
Next
```

This is a limitation in VB, not very often needed but being able to do that can open new possibilities.
In the VB6 add-in environment that's possible, but not for normal programs.

Inspired by [this question](https://www.vbforums.com/showthread.php?892793-Clone-Objects-with-Sub-Objects-!)

**Comment by WaynePhillipsEA** [comment]
This comes down to type library information.  In the 32-bit world you've got the TLBINF32 library that can potentially give you that level of access (but that library is not available for 64-bit going forward).

We should consider designing a Reflection API (similar to .NET), to provide access to runtime object information like this.
